test/integration/fake_upstream.h

```diff
@@ -343,13 +333,8 @@ class SharedConnectionWrapper : public Network::ConnectionCallbacks,
     if (!time_system.waitFor(lock_, absl::Condition(&callback_ready_event), timeout)) {
       return testing::AssertionFailure() << "Timed out while executing on dispatcher.";
     }
-    if (unexpected_disconnect && !allow_unexpected_disconnects_) {
-      return testing::AssertionFailure()
-             << "The connection disconnected unexpectedly, and allow_unexpected_disconnects_ is "
-                "false."
-                "\n See "
-                "https://github.com/envoyproxy/envoy/blob/master/test/integration/README.md#"
-                "unexpected-disconnects";
+    if (unexpected_disconnect && !allow_disconnects) {
+      ENVOY_LOG_MISC(warn, "executeOnDispatcher failed due to disconnect\n");
     }
```

Comment:
Is anyone ever going to see this? Delete also?

Reply:
My thought, and why I left it at warn, was if some executeOnDispatcher call asserts the call happens when in fact it's an unexpected disconnect, it'd be a pain to debug without at least some breadcrumb.

Reply:
OK SGTM. We can always iterate further later!